pkg/volumes/do/discovery.go

```diff
@@ -32,40 +32,30 @@ func (a *DOVolumes) Poll() (map[string]discovery.Node, error) {
 
 	glog.V(2).Infof("Polling all DO Volumes")
 
-	var mytag = a.ClusterName + "-" + a.nameTag
-
 	allVolumes, err := a.findVolumes(false)
 	if err != nil {
 		return nil, fmt.Errorf("failed to discover any volumes: %s", err)
 	}
 
-	// Iterate over every volume. By the time this is called, the instance should have the volume mounted
-	// When mounting a volume to an instance, the internal DOVolume object's Info's Description is set to the
-	// clustername_name tag
-	// which would either be k8s.io/etcd/main OR k8s.io/etcd/events.
-	// This name tag is passed as a command parameter to this container and the same is updated in DOVolumes
-	// Verify if the volume we are iterating matches the nameTag.
-	// This is a HACK until we update KOPS to utlize tags that we can compare here.
 	instanceToVolumeMap := make(map[string]*volumes.Volume)
 	for _, v := range allVolumes {
 		if v.AttachedTo != "" {
-			if mytag == v.Info.Description {
-
-				var myIP, _ = a.MyIP()
-
-				glog.V(2).Infof("Discovered a matching DO Volume for this instance with instanceid=%s volumename=%s nameTag=%s myIP=%s etcdName=%s", v.AttachedTo, v.ProviderID, a.nameTag, myIP, v.EtcdName)
+			instanceToVolumeMap[v.AttachedTo] = v
+		}
+	}
 
-				instanceToVolumeMap[v.AttachedTo] = v
+	for _, volume := range instanceToVolumeMap {
+		var myIP, _ = a.MyIP()
```

Comment:
The intent here is to find the other droplets that have mounted volumes, so we can start gossiping with them.  I don't think therefore that we want to filter only to the volumes that we have mounted.  For example, the AWS code has to do another call to get the instance ips here: https://github.com/kopeio/etcd-manager/blob/master/pkg/volumes/aws/discovery.go

Reply:
Thanks for mentioning this. I figured out this was an issue after adding protokube seed support for DO. With the seed implemented for DO and fixing the discovery code, I am now able to see the gossip cluster created Ok.
Below is the logs from /var/log/etcd.

```
 {"name":"kops-3-etcd-main-dev5-k8s-local","peerURLs":["https://kops-3-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:2380"],"endpoints":["https://kops-3-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:4001"],"ID":"3144370236654358750"}
   {"name":"kops-2-etcd-main-dev5-k8s-local","peerURLs":["https://kops-2-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:2380"],"endpoints":["https://kops-2-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:4001"],"ID":"10672077060946512164"}
   {"name":"kops-1-etcd-main-dev5-k8s-local","peerURLs":["https://kops-1-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:2380"],"endpoints":["https://kops-1-etcd-main-dev5-k8s-local.internal.dev5.k8s.local:4001"],"ID":"16336605654655619445"}
```